godot-core/src/engine/script_instance.rs

```diff
@@ -318,32 +259,189 @@ pub fn create_script_instance<T: ScriptInstance>(rs_instance: T) -> *mut c_void
     }
 }
 
+/// Mutable/exclusive reference guard for a `T` where `T` implements [`ScriptInstance`].
+///
+/// This can be used to access the base object of a [`ScriptInstance`], which in turn can be used to make reentrant calls to engine APIs.
+/// For details see [`SiMut::base_mut()`].
+pub struct SiMut<'a, T: ScriptInstance> {
+    mut_ref: &'a mut T,
+    cell: Pin<&'a GdCell<T>>,
+    base_ref: &'a Base<T::Base>,
+}
+
+impl<'a, T: ScriptInstance> SiMut<'a, T> {
+    fn new(
+        cell: Pin<&'a GdCell<T>>,
+        cell_guard: &'a mut MutGuard<T>,
+        base_ref: &'a Base<T::Base>,
+    ) -> Self {
+        let mut_ref = cell_guard.deref_mut();
+
+        Self {
+            mut_ref,
+            cell,
+            base_ref,
+        }
+    }
+
+    /// Returns a shared reference suitable for calling engine methods on this object.
+    ///
+    /// ```no_run
+    /// # use godot::prelude::*;
+    /// # use godot::engine::{ScriptInstance, SiMut, ScriptLanguage, Script};
+    /// # use godot::builtin::meta::{MethodInfo, PropertyInfo};
+    /// # use godot::sys;
+    ///
```

Comment:
Remove this newline please (also in other places).

You can check the rendering on https://godot-rust.github.io/docs/gdext/pr-671/godot/engine/struct.SiMut.html in near-realtime 🙂 

There's also still a newline after `}`.